storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1374,5 +1374,61 @@ public static RuntimeException wrapInRuntime(Exception e){
             return new RuntimeException(e);
         }
     }
+
+    public static <T> long bitXorValsSets(java.util.Set<T> vals) {
+        long rtn = 0l;
+        for (T n : vals) {
+            rtn = bitXor(rtn, n);
+        }
+        return rtn;
+    }
+
+    public static long bitXor(Object a, Object b) {
```

Comment:
This piece of code comes from JStorm. There is some purpose for extension in JStorm. But anyway, I will remove the extra checking.